### PR TITLE
[Merged by Bors] - feat(Order/WithBot, Data/Real/ENNReal): coe_injective

### PR DESCRIPTION
**Pull request by nomeata**
There already exists `NNReal.coe_injective` and
`WithBot.coe_injective`, so this adds the analogous
`ENNReal.coe_injective` and `WithTop.injective`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by jcommelin — APPROVED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

File: Mathlib/Order/WithBot.lean [hunk 1]
```diff
@@ -584,6 +584,15 @@ instance inhabited : Inhabited (WithTop α) :=
 instance nontrivial [Nonempty α] : Nontrivial (WithTop α) :=
   Option.nontrivial
 
+open Function
+
+theorem coe_injective : Injective (fun (a : α) => (a : WithTop α)) :=
+  Option.some_injective _
+
+@[simp, norm_cast]
```
Comment (nomeata) [inline Mathlib/Order/WithBot.lean:592]:
```suggestion
@[norm_cast]
```

Comment (nomeata) [reply Mathlib/Order/WithBot.lean:592]: No need to simp

File: Mathlib/Order/WithBot.lean [hunk 2]
```diff
@@ -584,6 +584,15 @@ instance inhabited : Inhabited (WithTop α) :=
 instance nontrivial [Nonempty α] : Nontrivial (WithTop α) :=
   Option.nontrivial
 
+open Function
+
+theorem coe_injective : Injective (fun (a : α) => (a : WithTop α)) :=
```
Comment (eric-wieser) [inline Mathlib/Order/WithBot.lean:589]:
```suggestion
theorem coe_injective : Injective ((↑) : α → WithTop α) :=
```
is a better term to work with in some cases

Comment (nomeata) [reply Mathlib/Order/WithBot.lean:589]: Ok, then I should do the same change to `WithBot`

**Pull request closed without merging**